build-system/tasks/compile-jison.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const fs = require('fs-extra');
+const glob = require('glob');
+const jison = require('jison');
+const path = require('path');
+const {endBuildStep} = require('./helpers');
+const {jisonPaths} = require('../config');
+
+// set imports for each parser from directory build/parsers/.
+const imports = new Map();
+imports.set(
+  'cssParser',
+  "import * as ast from '../../extensions/amp-animation/0.1/parsers/css-expr-ast';"
+);
+imports.set(
+  'bindParser',
+  "import {AstNode, AstNodeType} from '../../extensions/amp-bind/0.1/bind-expr-defines';\n" +
+    "import {tryParseJson} from '../../src/json';"
+);
+
+/**
+ * Builds parsers for extensions with *.jison files.
+ * Uses jison file path to name the parser to export.
+ * For example, css-expr-impl.jison creates `cssParser`.
+ * @return {!Promise<void>}
```

Comment:
just curious, is `void` an actual type?

Reply:
I guess it's the lack of types. I see this in typescript.